### PR TITLE
dnsdist: Add RewriteMapAction and RewriteMapResponseAction

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

Add `RewriteMapAction` to rewrite the qname of queries based on the content of the map. For example, if the map contains a key `example.com.` with the associated value `notexample.org.`, a query for `test.example.com.` will be sent to the backend with a QName of `test.notexample.org.`.
To function properly, a matching `RewriteMapResponseAction` action should be defined to rewrite the names and contents in responses. Only the following record types see their content rewritten, others are left untouched:
* CNAME
* MX
* NS
* SRV

Remaining questions:
* do we need to handle a query for zone `a` getting a record pointing to a name in zone `b`, where both `a` and `b` are in the map?
* do we want to handle the case where a query for a zone not present in the map receives a record pointing to a zone that is in the map?

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/dnsdist-lua-actions.cc [hunk 1]
```diff
@@ -923,6 +924,189 @@ class TagResponseAction : public DNSResponseAction
...
+  DNSName rebased(*dq->qname);
+  rebased.rebase(pair->first, pair->second);
+  char* buffer = reinterpret_cast<char*>(dq->dh);
+  const size_t originalQNameSize = dq->qname->wirelength();
+  const std::string newQNameWire =  rebased.toDNSString();
+  const size_t newQNameSize = newQNameWire.size();
+  size_t newSize = dq->len - originalQNameSize + newQNameSize;
+
+  if (dq->len < originalQNameSize || newSize > dq->size) {
+    return Action::None;
```
Comment (zeha) [inline pdns/dnsdist-lua-actions.cc]: some form of warnlog here?

File: pdns/dnsdist-lua-actions.cc [hunk 2]
```diff
@@ -923,6 +924,189 @@ class TagResponseAction : public DNSResponseAction
...
+  catch(const std::exception& e) {
+    vinfolog("Got exception while rewriting answer: %s", e.what());
+  }
+
+  return false;
+}
+
+DNSResponseAction::Action RewriteMapResponseAction::operator()(DNSResponse* dr, string* ruleresult) const
+{
+  const auto pair = d_smt.lookup(*dr->qname);
```
Comment (zeha) [inline pdns/dnsdist-lua-actions.cc]: could have a nicer name than `pair` here. maybe `fromTo`?

**Comment by rgacogne** [comment]
Moving this to 1.4.0 since we still have open questions on that feature.

**Comment by rgacogne** [comment]
Closing for lack of interest!

**Pull request closed without merging**